src/NUnit.TestAdapter.Tests.Acceptance/TestSourceWithCustomNames.cs

```diff
@@ -0,0 +1,130 @@
+﻿using System.Linq;
+using NUnit.Framework;
+using NUnit.VisualStudio.TestAdapter.Tests.Acceptance.WorkspaceTools;
+
+namespace NUnit.VisualStudio.TestAdapter.Tests.Acceptance
+{
+    public sealed class TestSourceWithCustomNames : AcceptanceTests
+    {
+        private static void AddTestsCs(IsolatedWorkspace workspace)
+        {
+            workspace.AddFile("Tests.cs", @"
+                using System;
+                using System.Collections;
+                using System.Collections.Generic;
+                using NUnit.Framework;
+
+                namespace Test
+                {
+                    public class Tests
+                    {
+                        [Test]
+                        [TestCaseSource(nameof(TestCaseSourceMethod))]
+                        public void PassingTestStr(object arg)
+                        {
+                            Assert.Pass();
+                        }
+
+
+                        private static IEnumerable<TestCaseData> TestCaseSourceMethod()
+                        {
+                            yield return new TestCaseData(""Name with mismatched parenthesis 'pp:-) :-)'"").SetName(""Name with mismatched parenthesis 'pp:-) :-)'"");
+                            yield return new TestCaseData(""Name with mismatched quote '\""c'"").SetName(""Name with mismatched quote '\""c'"");
+                            
+                            // Adding a parenthesis to the end of this test name will stop the exception from throwing (e.g. $""TestName(...)"")
+                            yield return new TestCaseData(1).SetName($""TestName(..."");
+
+                            // Cannot be duplicated without a second test included that ends with a ']'
+                            yield return new TestCaseData(2).SetName($""TestName(...)]"");
+                        }
+
+                        [Test]
+                        [TestCase(typeof(IEnumerable<(String oneValue, Int32 twoValue)>))]
```

Comment:
What is the reason for using `String` and `Int32` here when `string` and `int` have been used in the rest of the class?

Reply:
Oh, that was unintended and is an artifact of me copy&pasting it from another tool while I worked on it.  I'll update them in my next commit.